src/App/contexts/taskContext.js

```diff
@@ -66,14 +101,38 @@ export const TaskProvider = (props) => {
     }
   }
 
+  // modify data
+  const toggleAllTasks = () => {
+    if (taskData.find(checkUncompleted)) {
+      taskData.forEach((task) => {
+        updateData({
+          id: task._id,
+          isCompleted: true,
+        });
+      });
+    } else {
+      taskData.forEach((task) => {
+        updateData({
+          id: task._id,
+          isCompleted: false,
+        });
+      });
+    }
```

Comment:
Why did you have to split if else like this?
And what type of data that taskData.find(checkUncompleted)) return? I assume it is an object, right? 
What type of `checkUncompleted`? Do you understand how **Array.prototype.find()** works?
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/find
```
const isCompleted = ... // The way you check uncompleted or not
taskData.forEach(({_id: id}) => {
  updateData({id, isCompleted})
})
```

Reply:
In checkUncompleted, it will return the first task with isCompleted = false, if it exist in array then I will update all task. I will change this function according to your advice